include/ck_tile/core/arch/utility.hpp

```diff
@@ -117,6 +117,45 @@ CK_TILE_DEVICE T warp_shuffle(const T& v_local, uint32_t src_lane)
 #endif
 }
 
+/// Make \p v_local uniform for all lanes in the warp and store it in scalar registers (SGPR).
+/// The value of \p v_local is read from the first active lane using v_readfirstlane_b32.
+/// No-op if the value is already stored in scalar registers.
+/// The type \tparam T must be trivially copyable.
+template <typename T>
+CK_TILE_DEVICE T warp_uniform(const T& v_local)
```

Comment:
I think `amd_wave_read_first_lane` in #2913 does the same thing. Please unify this functionality if possible.

Reply:
Never mind, my bad, I misunderstood that change: in that PR `amd_wave_read_first_lane` is defined in both places so it should work on older and newer compilers.

Ok, I can wait for that PR if it's about to be merged (or I can do as a separate fix later if this PR is needed sooner)